frontend/src/features/rubricBuilder/CSVUpload.tsx

```diff
@@ -42,30 +58,201 @@ export const CSVUpload: React.FC<CSVUploadProps> = ({
               )
           );
         });
-        console.log("Validated CSV data:", results.data);
-        onDataChange(parsedData); // Pass validated data to parent
+
+        updateRubricWithCSVVersion1(parsedData);
+      },
+    });
+  };
+
+  const updateRubricWithCSVVersion1 = (data: CSVRow[]) => {
+    if (!rubric) return;
```

Comment:
For all these variable names, I think CSV operations are implied enough in the component

Could just be `updateVersionOne`, `parseVersionTwo`, etc. for readability.